src/main/java/domain/Lottos.java

```diff
@@ -1,21 +1,23 @@
 package domain;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
-import domain.Lotto;
-
 public class Lottos {
 	private final List<Lotto> lottos;
 
 	public Lottos(List<Lotto> lottos) {
 		this.lottos = lottos;
 	}
 
-	public int getLottoCount() {
-		return lottos.size();
+	public List<Lotto> getLottos() {
+		return Collections.unmodifiableList(lottos);
 	}
 
-	public List<Lotto> getLottos() {
-		return lottos;
+	public WinningResult calculateWinningResult(WinningLotto winningLotto) {
+		WinningResult result = new WinningResult();
+		getLottos().forEach(lotto -> result.add(winningLotto.getRank(lotto)));
+		return result;
```

Comment:
WinningResult 객체 생성 후, 값을 하나씩 더해가는 방식보다는 
WinningResult 객체가 자신의 역할을 곧바로 수행할 수 있는 상태로 바로 초기화해주면 WinningResult는 자신의 역할을 수행하는데만 온전히 집중할 수 있을 것 같아요. 😃

```suggestion
		List<Rank> ranks = this.lottos.stream()
                    .map(winningLotto::getRank)
                    .collect(Collectors.toList());
                return new WinningResult(ranks);
```

Reply:
`List<Rank>` 타입 인자로 받는 `WinningResult` 생성자 추가했습니다~ 생성 시점에 온전한 객체가 되어야한다 좋은 말이네욥.